relay-event-schema/src/protocol/trace_metric.rs

```diff
@@ -0,0 +1,186 @@
+use relay_protocol::{
+    Annotated, Empty, FromValue, Getter, IntoValue, Object, SkipSerialization, Value,
+};
+use std::fmt::{self, Display};
+
+use relay_base_schema::metrics::MetricUnit;
+use serde::{Serialize, Serializer};
+
+use crate::processor::ProcessValue;
+use crate::protocol::{Attributes, SpanId, Timestamp, TraceId};
+
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[metastructure(process_func = "process_trace_metric", value_type = "TraceMetric")]
+pub struct TraceMetric {
+    /// Timestamp when the metric was created.
+    #[metastructure(required = true)]
+    pub timestamp: Annotated<Timestamp>,
+
+    /// The ID of the trace the metric belongs to.
+    #[metastructure(pii = "false", required = true, trim = false)]
+    pub trace_id: Annotated<TraceId>,
+
+    /// The Span this metric belongs to.
+    #[metastructure(pii = "false", required = false, trim = false)]
+    pub span_id: Annotated<SpanId>,
+
+    /// The metric name.
+    #[metastructure(pii = "false", required = true, trim = false)]
+    pub name: Annotated<String>,
+
+    /// The metric type.
+    #[metastructure(pii = "false", required = true, field = "type")]
+    pub r#type: Annotated<MetricType>,
+
+    #[metastructure(pii = "false", required = false)]
+    pub unit: Annotated<MetricUnit>,
+
+    /// The metric value. Should be constrained to a number.
+    #[metastructure(pii = "maybe", required = true, trim = false)]
+    pub value: Annotated<Value>,
+
+    /// Arbitrary attributes on a metric.
+    #[metastructure(pii = "maybe", trim = false)]
+    pub attributes: Annotated<Attributes>,
+
+    /// Additional arbitrary fields for forwards compatibility.
+    #[metastructure(additional_properties, retain = true, pii = "maybe")]
```

Comment:
With `retain=false`, external relays will forward new fields, but processing relays will drop them. I believe that is what we want here.

```suggestion
    #[metastructure(additional_properties, retain = false)]
```